news/refactor-main.rst

```diff
@@ -0,0 +1,23 @@
+**Added:**
+
+* <news item>
+
+**Changed:**
+
+* Rewrite main() with the refactored functions.
```

Comment:
this is a bit cryptic.  It makes sense to you in the context of what you are doing now, but in a CHANGELOG no-one will really follow what you mean.